testing/testcontainers/java/README.rst

```diff
@@ -0,0 +1,63 @@
+#######################
+Testcontainers for Java
+#######################
+
+*How to run integration tests of Java applications with CrateDB.*
+
+
+*****
+About
+*****
+
+Introduction
+============
+
+`Testcontainers for Java`_ is a Java library that supports JUnit tests,
+providing lightweight, throwaway instances of common databases suitable
+for integration testing scenarios.
+
+The `Testcontainers CrateDB Module`_ will provide your application test
+framework with a single-node CrateDB instance. You will be able to choose
+the `CrateDB OCI image`_ variant by version, or by selecting the ``nightly``
+release.
+
+What's inside
+=============
+
+This directory includes different canonical examples how to use those
+components within test harnesses of custom applications. Currently,
+all test cases are based on JUnit 4.
+
+
+*****
+Usage
+*****
+
+1. Make sure Java 17 is installed.
+2. Run CrateDB::
+
+       docker run -it --rm --publish=4200:4200 --publish=5432:5432 \
+           crate:latest -Cdiscovery.type=single-node
+
+3. Invoke example application::
+
+    ./gradlew run --args="jdbc:crate://localhost:5432/"
+    ./gradlew run --args="jdbc:postgresql://localhost:5432/"
+
+4. Invoke software tests::
```

Comment:
Maybe want to point out that we don't need to run crate docker container for the tests, it's being spanwed by testcontainers. Additionally, if we have started it already (step 2), would there be any port conflicts when we run the tests?

Reply:
Thanks. I've swapped the order of both sections, because running the test cases should be more prominent. There will not be any port conflicts, because Testcontainers-driven instances will be assigned random ports.

Reply:
Thx!